File: projects/gmail-outreach/fetch-sheet-cron.js
const { google } = require('googleapis');
const fs = require('fs');

const SHEET_ID = '11TRs92xmRWJ_FEQ_0nnLDrUkPPJRSqTG_iBSYBjGov4';
const SERVICE_ACCOUNT_FILE = 'service-account.json';

async function fetchSheet() {
  const auth = new google.auth.GoogleAuth({
    keyFile: SERVICE_ACCOUNT_FILE,
    scopes: ['https://www.googleapis.com/auth/spreadsheets.readonly'],
  });
  
  const sheets = google.sheets({ version: 'v4', auth });

  const response = await sheets.spreadsheets.values.get({
    spreadsheetId: SHEET_ID,
    range: 'Sheet1!A:I',
  });
  
  const rows = response.data.values || [];
  const header = rows[0];
  
  console.log('Header:', header);
  console.log(`\nTotal rows: ${rows.length - 1}\n`);
  
  // Find column indices
  const companyCol = header.indexOf('Company');
  const contactCol = header.indexOf('Contact Name');
  const titleCol = header.indexOf('Position/Title');
  const emailCol = header.indexOf('Email');
  const statusCol = header.indexOf('Status');
  const notesCol = header.indexOf('Notes');

  // Find rows needing enrichment
  const needsEnrichment = [];
  
  for (let i = 1; i < rows.length; i++) {
    const row = rows[i];
    const company = row[companyCol] || '';
    const contact = row[contactCol] || '';
    const email = row[emailCol] || '';
    const status = row[statusCol] || '';
    
    // Check if needs enrichment
    const noContact = !contact || contact.trim() === '';
    const genericEmail = email && (
      email.toLowerCase().includes('info@') || 
      email.toLowerCase().includes('sales@') ||
      email.toLowerCase().includes('ir@') ||
      email.toLowerCase().includes('contact@')
    );
    
    if (company && (noContact || genericEmail) && status !== 'Dead' && !status.includes('Dead')) {
      needsEnrichment.push({
        row: i + 1,
        company,
        contact,
        email,
        status,
        reason: noContact ? 'No contact' : 'Generic email'
      });
    }
  }
  
  console.log(`\n📊 Rows needing enrichment: ${needsEnrichment.length}\n`);
  
  // Show first 15
  const toShow = needsEnrichment.slice(0, 15);
  toShow.forEach(item => {
    console.log(`Row ${item.row}: ${item.company}`);
    console.log(`  Contact: ${item.contact || '(empty)'}`);
    console.log(`  Email: ${item.email || '(empty)'}`);
    console.log(`  Status: ${item.status || '(empty)'}`);
    console.log(`  Reason: ${item.reason}`);
    console.log('');
  });
  
  // Save to file for processing
  fs.writeFileSync('enrichment-targets-march5-506pm.json', JSON.stringify(needsEnrichment, null, 2));
  console.log(`\n✅ Saved ${needsEnrichment.length} targets to enrichment-targets-march5-506pm.json`);
}

fetchSheet().catch(console.error);

File: projects/gmail-outreach/fetch-sheet-cron.py
#!/usr/bin/env python3
import json
from google.oauth2 import service_account
from googleapiclient.discovery import build

SHEET_ID = '11TRs92xmRWJ_FEQ_0nnLDrUkPPJRSqTG_iBSYBjGov4'
SERVICE_ACCOUNT_FILE = 'service-account.json'
SCOPES = ['https://www.googleapis.com/auth/spreadsheets.readonly']

def fetch_sheet():
    creds = service_account.Credentials.from_service_account_file(
        SERVICE_ACCOUNT_FILE, scopes=SCOPES)
    
    service = build('sheets', 'v4', credentials=creds)
    sheet = service.spreadsheets()
    
    result = sheet.values().get(
        spreadsheetId=SHEET_ID,
        range='Sheet1!A:I'
    ).execute()
    
    rows = result.get('values', [])
    
    if not rows:
        print('No data found.')
        return
    
    header = rows[0]
    print(f'Header: {header}')
    print(f'\nTotal rows: {len(rows) - 1}\n')
    
    # Find column indices
    company_col = header.index('Company') if 'Company' in header else -1
    contact_col = header.index('Contact Name') if 'Contact Name' in header else -1
    title_col = header.index('Position/Title') if 'Position/Title' in header else -1
    email_col = header.index('Email') if 'Email' in header else -1
    status_col = header.index('Status') if 'Status' in header else -1
    notes_col = header.index('Notes') if 'Notes' in header else -1
    
    # Find rows needing enrichment
    needs_enrichment = []
    
    for i in range(1, len(rows)):
        row = rows[i]
        
        # Pad row to match header length
        while len(row) < len(header):
            row.append('')
        
        company = row[company_col] if company_col >= 0 else ''
        contact = row[contact_col] if contact_col >= 0 else ''
        email = row[email_col] if email_col >= 0 else ''
        status = row[status_col] if status_col >= 0 else ''
        
        # Check if needs enrichment
        no_contact = not contact or contact.strip() == ''
        generic_email = email and any(x in email.lower() for x in ['info@', 'sales@', 'ir@', 'contact@'])
        
        if company and (no_contact or generic_email) and status != 'Dead' and 'Dead' not in status:
            needs_enrichment.append({
                'row': i + 1,
                'company': company,
                'contact': contact,
                'email': email,
                'status': status,
                'reason': 'No contact' if no_contact else 'Generic email'
            })
    
    print(f'\n📊 Rows needing enrichment: {len(needs_enrichment)}\n')
    
    # Show first 15
    to_show = needs_enrichment[:15]
    for item in to_show:
        print(f"Row {item['row']}: {item['company']}")
        print(f"  Contact: {item['contact'] or '(empty)'}")
        print(f"  Email: {item['email'] or '(empty)'}")
        print(f"  Status: {item['status'] or '(empty)'}")
        print(f"  Reason: {item['reason']}")
        print('')
    
    # Save to file
    with open('enrichment-targets-march5-506pm.json', 'w') as f:
        json.dump(needs_enrichment, f, indent=2)
    
    print(f'\n✅ Saved {len(needs_enrichment)} targets to enrichment-targets-march5-506pm.json')

if __name__ == '__main__':
    fetch_sheet()
